neat/cli/commands/parallel.py

```diff
@@ -0,0 +1,100 @@
+
+
+class Command(BaseCommand):
+    """
+    Split the reference, run read-simulator, and stitch outputs together.
+    """
+    name = "parallel"
+    description = "Split the reference, run read-simulator, and stitch outputs together."
+
+    def add_arguments(self, parser: argparse.ArgumentParser) -> None:
+        """
+        Add the command's arguments to its parser
+
+        :param parser: The parser to add arguments to
+        """
+        parser.add_argument("-c",
+                            "--config",
+                            type=Path,
+                            required=True,
+                            help="NEAT YAML/YML config containing the 'reference:' field")
+
+        parser.add_argument("--outdir",
+                            type=Path,
+                            required=False,
+                            default=None,
+                            help="Top-level directory for splits and stitched results (optional)")
+
+        split = parser.add_argument_group("splitting options")
+        split.add_argument("--by", choices=["contig", "size"], default=None,
+                           help="Split mode")
+        split.add_argument("--size", type=int, default=None,
+                           help="Target chunk size when --by size")
+        split.add_argument("--cleanup-splits", action=argparse.BooleanOptionalAction, default=None,
+                           help="Delete the 'splits' directory after stitching completes")
+        split.add_argument("--reuse-splits", action=argparse.BooleanOptionalAction, default=None,
+                           help="Skip splitting and reuse existing YAML/FASTA files in 'splits'")
+
+        sim = parser.add_argument_group("simulation options")
+        sim.add_argument("--jobs", type=int, default=None,
+                         help="Maximum number of parallel NEAT jobs")
+        sim.add_argument("--neat-cmd", default=None,
+                         help="Command used to launch the read simulator (e.g., 'neat read-simulator')")
+
+        stitch = parser.add_argument_group("stitching options")
+        stitch.add_argument("--samtools", default=None,
+                            help="Path to samtools executable used by stitch_outputs.py")
+        stitch.add_argument("--final-prefix", type=Path, default=None,
+                            help="Prefix (no extension) for stitched outputs")
+
+        parser.add_argument("--parallel-config", type=Path,
+                            help="Optional YAML/JSON file with parallelization settings (jobs, by, size, etc.)")
+
+    def execute(self, arguments: argparse.Namespace) -> None:
+        # Optionally overlay values from a parallel-config file
+        if arguments.parallel_config and arguments.parallel_config.is_file():
+            import json, yaml
+            ext = arguments.parallel_config.suffix.lower()
+            with open(arguments.parallel_config, "r") as fh:
```

Comment:
Check out some of the file functions we have written in common/io.py -> open files, write files etc. Includes some safety checks etc.